src/main/java/org/opentripplanner/routing/graph/GraphIndex.java

```diff
@@ -516,6 +516,119 @@ protected boolean lessThan(TripTimeShort tripTimeShort, TripTimeShort t1) {
     }
 
     /**
+     * Fetch next n upcoming vehicle departures for a stop of pattern. It goes
+     * though the previous, current and next service date. It uses a priority
+     * queue to keep track of the next departures. The queue is shared between
+     * all dates, as services from the previous service date can visit the stop
+     * later than the current service date's services. This happens eg. with
+     * sleeper trains.
+     *
+     * @param stop
+     *            Stop object to perform the search for
+     * @param startTime
+     *            Start time for the search. Seconds from UNIX epoch
+     * @param TripPattern
+     *            The selected pattern. If null an empty list is returned.
+     * @param timeRange
+     *            Searches forward for timeRange seconds from startTime
+     * @param numberOfDepartures
+     *            Number of departures to fetch per pattern
+     * @return
+     */
+    public List<TripTimeShort> stopTimesForPattern(Stop stop, TripPattern pattern, long startTime, int timeRange,
```

Comment:
There is quit much code shared with `stopTimesForStop` would it make sense to refactor some of that into common method(s)?